common/firmware/adc/adc.c

```diff
@@ -0,0 +1,155 @@
+#include "common/firmware/adc.h"
```

Comment:
This C file is entirely specific to the hardware of the head project. It configures the ADC in a way that is only relevant to head, it maps pinmux in a way that is only relevant to head, and it defines helper functions like `read_voltage_[gripper|a_motor|z_motor]` that are not only specific to head in implementation but specific to head in semantics. This file and its header really need to either in the `head` subproject.